### ISSUE TITLE
Python 2 package incorrectly installed over Python 3 version

### ISSUE THREAD

**Issue by laactech** [sdispater/poetry#877]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Ubuntu 18.04
- **Poetry version**: 1.0.0a2
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: 
```toml
[tool.poetry]
name = "poetry_bug"
version = "0.1.0"
description = ""
authors = ["Steven Pate <steven@laac.io>"]

[tool.poetry.dependencies]
python = "^3.6"
django = "^2.1"

[tool.poetry.dev-dependencies]
pytest = "^4.2"

[build-system]
requires = ["poetry>=0.12"]
build-backend = "poetry.masonry.api"
```

## Issue

Once again, thanks for your work on poetry. I really enjoy using it.

Using the above pyproject.toml, I run `poetry add django-allauth` which outputs this:
```
Writing lock file

  - Installing certifi (2018.11.29)
  - Installing chardet (3.0.4)
  - Installing idna (2.8)
  - Installing urllib3 (1.24.1)
  - Installing oauthlib (3.0.1)
  - Installing requests (2.21.0)
  - Installing python-openid (2.2.5)
  - Installing requests-oauthlib (1.2.0)
  - Installing django-allauth (0.38.0)
```
The key part is `Installing python-openid (2.2.5)`. I'm using `python 3.6`, and my expectation is that `python3-openid` would be installed. Here's the relevant line from django-allauth's setup.py.

https://github.com/pennersr/django-allauth/blob/master/setup.py#L106

Any idea why the Python 2 version is being installed? I'm not sure if this is on the poetry end or the django-allauth end.

**Comment by vphilippon** [comment]
I took a look at the `.tar.gz` published, I might have an idea that `poetry` maintainer will be able to confirm.

In the `setup.py`, there's this segment:
```
IS_PY2 = sys.version_info[0] < 3

if IS_PY2:
    openid_package = 'python-openid >= 2.2.5'
    test_requirements.append('mock >= 1.0.1')
else:
    openid_package = 'python3-openid >= 3.0.8'

# ... skipping some code ...

    install_requires=['Django >= 1.11',
                               openid_package,  # My note: The variable defined above.
                               'requests-oauthlib >= 0.3.0',
                               "requests"],
```

But in `django_allauth.egg-info/requires.txt` there's:
```
Django>=1.11
python-openid>=2.2.5
requests-oauthlib>=0.3.0
requests
```

So in short, the metadata from `django_allauth`'s `requires.txt` simply list the Py2 variant of the dependencies, not the Py3 ones. A side effect of having custom code in `setup.py` which depends on which version of python used.

I _think_ that `poetry` relies on the information in the `.egg-info` rather than executing the `setup.py` as `pip` does. @sdispater Could you confirm this?

`django_allauth` maintainers could address this by using some standard [environment markers](https://www.python.org/dev/peps/pep-0508/#environment-markers) in their `install_requires` list instead of some custom code in `setup.py` which relies on `setup.py` always being executed (the World is moving away from that assumption, slowly).

TL;DR `django_allauth` maintainers should probably update their packaging practices to some newer, better practices.

**Comment by bertjwregeer** [comment]
The authors of django-allauth need to use environment markers instead of custom code in setup.py.

**Comment by laactech** [comment]
Closing since this is not a poetry issue.
